### PR TITLE
Switch to eck-diagnostics 1.0.1 and remove eckdump.sh

### PR DESCRIPTION
**Pull request by pebrc**
This updates the CI Docker image to use the latest eck-diagnostics build and replaces the `eckdump.sh` script with a link to the new tooling.

### CODE REVIEW

File: hack/diagnostics/README.md [hunk 1]
```diff
@@ -1 +1 @@
-This resource has been moved to a [new location](../../support/diagnostics/) within the top level [support](../../support) folder.
+Please find the support diagnostics tool for ECK at https://github.com/elastic/eck-diagnostics
```
Comment (thbkrkr) [inline hack/diagnostics/README.md:1]:
```suggestion
Please find the support diagnostics tool for ECK at https://github.com/elastic/eck-diagnostics.
```

File: support/diagnostics/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Please find the support diagnostics tool for ECK at https://github.com/elastic/eck-diagnostics
```
Comment (thbkrkr) [inline support/diagnostics/README.md:1]:
```suggestion
Please find the support diagnostics tool for ECK at https://github.com/elastic/eck-diagnostics.
```

**Review by thbkrkr — COMMENTED** [review]
We also need to update [docs/operating-eck/troubleshooting/take-eck-dump.asciidoc](https://www.elastic.co/guide/en/cloud-on-k8s/1.8/k8s-take-eck-dump.html). Do you want to do it in another PR or in this one?

**Comment by pebrc** [comment]
> We also need to update docs/operating-eck/troubleshooting/take-eck-dump.asciidoc. Do you want to do it in another PR or in this one?

Ah I forgot that we have that page now. I will push an additional commit to address that.

**Pull request merged**